templates/default/vlan/vlanmodify-dialog.html

```diff
@@ -79,11 +79,44 @@
         title: "{if $action == 'modify'}{trans("Modify VLAN")} {$vlaninfo.id}{else}{trans("Add VLAN")}{/if}"
     });
 
-    $( "#addvlanbtn" ).click(function() {
-        $( "#addvlanmodal" ).dialog( "open" );
-    });
-
     $( "#close" ).click(function() {
         $( "#addvlanmodal" ).dialog( "close" );
     });
+
+    function open_add_dialog() {
+        $( "#addvlan-form" ).attr('action', '?m=vlanlist&action=add');
+        $( "#addvlanmodal" ).dialog( "open" );
+        $( "#addvlanmodal" ).dialog( "option", "title", "{trans("Add VLAN")}");
+
+        document.getElementById("dialog-vlanid").value = "";
+        document.getElementById("dialog-description").value = "";
+        document.getElementById("dialog-netnode_selector").value = "";
+        updateAdvancedSelects("#dialog-netnode_selector");
+        ///tutaj potrzebuję wyczyścić wybór klienta - początek
+        $( "#dialog-customer_selector" ).attr('value','');
+        $( "#dialog-customer_selector" ).attr('data-customer-name','');
+        $( "#dialog-customer_selector" ).attr('data-prev-value','');
+        $( ".lms-ui-customer-select-name").html('<a href=""></a>');
+        //tutaj potrzebuję wyczyścicć wybór klienta - koniec
+    };
+
+    function open_modify_dialog (template_id) {
+        $( "#addvlanmodal" ).dialog( "open" );
+        $( "#addvlanmodal" ).dialog( "option", "title", "{trans("Modify VLAN")} " + template_id);
+        $( "#addvlan-form" ).attr('action', '?m=vlanlist&action=modify&id=' + template_id);
+        if (template_id) {
+            var row = $('[data-template-id=' + template_id + ']');
+            $("#dialog-vlanid").val($(row).find("[data-template-vlanid]").attr('data-template-vlanid'));
+            $("#dialog-description").val($(row).find("[data-template-description]").attr('data-template-description'));
+            $("select#dialog-netnode_selector").val( $(row).find("[data-template-netnodeid]").attr('data-template-netnodeid'));
+            updateAdvancedSelects("#dialog-netnode_selector");
+
+            var customeridedit = $(row).find("[data-template-customerid]").attr('data-template-customerid');
+            $( "#dialog-customer_selector" ).attr('value', customeridedit);
+            $( "#dialog-customer_selector" ).attr('data-customer-name', customeridedit);
+            $( "#dialog-customer_selector" ).attr('data-prev-value', customeridedit);
+            $( ".lms-ui-customer-select-name" ).trigger('change');
```

Comment:
Ale to raczej nie to - trzeba szukać elementu `.lms-ui-customer-select-customerid` i na nim uruchomić `change()`.

Reply:
Jeszcze jedno to również na tym co podałem robisz:
```js
$('.lms-ui-customer-select-customerid').val(customeridedit).change();
```
lub
```js
$('.lms-ui-customer-select-customerid').val(customeridedit).trigger('change');
```

Reply:
Jak masz ustawione `phpui.big_networks`?

Reply:
Na instancji testowej `phpui.big_networks=false` bo to jest domyślnie w instancji.

Reply:
Podmieniam linię 118 na ten kod co tutaj napisałeś - obie nie aktualizują wartości w `{customerlist}`. W konsoli przeglądarki pusto.

Reply:
Jak wygląda pełny kod po zmianach?

Reply:
Raczej:
```js
$('.lms-ui-customer-select-customerid').val(customeridedit).trigger('change');
```
jest poprawnym wyborem.

Reply:
> Jak wygląda pełny kod po zmianach?

https://github.com/lmsgit/lms/pull/2023/files

Reply:
Testuje na `phpui.big_networks` false oraz true

Reply:
> Raczej:
> ```js
> $('.lms-ui-customer-select-customerid').val(customeridedit).trigger('change');
> ```
> jest poprawnym wyborem.

Duch mój na rozżarzone węgle gołą d*** usiadł. Nie wiem co mam poradzić z tym kodem JS. Nie działa.
@chilek pomożesz z tym?